spacy/language.py

```diff
@@ -1300,7 +1300,7 @@ def evaluate(
             kwargs.setdefault("batch_size", batch_size)
             # non-trainable components may have a pipe() implementation that refers to dummy
             # predict and set_annotations methods
-            if not hasattr(pipe, "pipe") or not hasattr(pipe, "is_trainable") or not pipe.is_trainable():
+            if not hasattr(pipe, "pipe"):
```

Comment:
Should we move this conditional into the `_pipe` function, and move that function to `spacy.util`?